standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -528,6 +533,7 @@ public static void updateTableStatsForCreateTable(Warehouse wh, Database db, Tab
         StatsSetupConst.ColumnStatsSetup statsSetup = StatsSetupConst.ColumnStatsSetup.parseStatsSetup(val);
         if (statsSetup.enabled) {
           try {
+            isIcebergTable = statsSetup.isIcebergTable;
             // For an Iceberg table, a new snapshot is generated, so any leftover files would be ignored
             // Set the column stats true in order to make it merge-able
             if (newDir || statsSetup.isIcebergTable ||
```

Comment:
If the current table is an Iceberg, then it will populate all the fast stats into table,

https://github.com/apache/hive/blob/6b095edc31a9baa0ae805e2b5aea88101284d84b/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java#L539

https://github.com/apache/hive/blob/6b095edc31a9baa0ae805e2b5aea88101284d84b/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/common/StatsSetupConst.java#L401-L405

As a result, the pre-condition in `updateTableStatsSlow` should be true, will ignore iterating the files under the table's directory.
https://github.com/apache/hive/blob/6b095edc31a9baa0ae805e2b5aea88101284d84b/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java#L453-L456

am I missing something?

Reply:
This is true when not creating Iceberg table as select.
But when CTAS like in this [q-test](https://github.com/apache/hive/blob/master/iceberg/iceberg-handler/src/test/queries/positive/ctas_iceberg_partitioned_orc.q), it will list all the files.
This is because when creating Iceberg table as select the table has no dictionary and this condition is false:
https://github.com/apache/hive/blob/e982ed1105062648b395c320425ea6683794f4d6/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java#L522
`    if (tbl.isSetDictionary() && tbl.getDictionary().getValues() != null) {`